services/src/main/java/org/fao/geonet/api/records/MetadataIndexApi.java

```diff
@@ -0,0 +1,155 @@
+import org.fao.geonet.ApplicationContextHolder;
+import org.fao.geonet.api.API;
+import org.fao.geonet.api.ApiParams;
+import org.fao.geonet.api.ApiUtils;
+import org.fao.geonet.api.exception.ResourceNotFoundException;
+import org.fao.geonet.api.exception.WebApplicationException;
+import org.fao.geonet.api.processing.XslProcessUtils;
+import org.fao.geonet.api.processing.report.XsltMetadataProcessingReport;
+import org.fao.geonet.api.records.model.suggestion.SuggestionType;
+import org.fao.geonet.api.records.model.suggestion.SuggestionsType;
+import org.fao.geonet.api.tools.i18n.LanguageUtils;
+import org.fao.geonet.domain.Metadata;
+import org.fao.geonet.exceptions.BadParameterEx;
+import org.fao.geonet.kernel.DataManager;
+import org.fao.geonet.kernel.SelectionManager;
+import org.fao.geonet.kernel.schema.MetadataSchema;
+import org.fao.geonet.kernel.setting.SettingManager;
+import org.fao.geonet.services.metadata.BatchOpsMetadataReindexer;
+import org.fao.geonet.utils.Xml;
+import org.jdom.Element;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.ApplicationContext;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.*;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.xml.transform.TransformerConfigurationException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import static org.fao.geonet.api.ApiParams.*;
+
+@RequestMapping(value = {
+    "/api/records",
+    "/api/" + API.VERSION_0_1 +
+        "/records"
+})
+@Api(value = API_CLASS_RECORD_TAG,
+    tags = API_CLASS_RECORD_TAG,
+    description = API_CLASS_RECORD_OPS)
+@Controller("recordIndexing")
+@ReadWriteController
+public class MetadataIndexApi {
+
+    @Autowired
+    DataManager dataManager;
+
+    @ApiOperation(
+        value = "Index selection",
+        notes = "Index selected metadata",
+        nickname = "indexSelection")
+    @RequestMapping(
+        value = "/selection/index",
+        method = RequestMethod.GET,
+        produces = MediaType.APPLICATION_JSON_VALUE
+    )
+    @PreAuthorize("hasRole('Administrator')")
+    @ResponseStatus(HttpStatus.OK)
+    @ApiResponses(value = {
+        @ApiResponse(code = 200, message = "Record indexed."),
+        @ApiResponse(code = 403, message = ApiParams.API_RESPONSE_NOT_ALLOWED_CAN_EDIT)
+    })
+    public
+    @ResponseBody
+    JSONObject index(
+            @ApiParam(
+                    value = ApiParams.API_PARAM_BUCKET_NAME,
+                    required = false)
+            @RequestParam(
+                    required = false
+            )
+                    String bucket,
```

Comment:
Maybe you could add the same parameters as in other services which apply to a set of records - could be defined by UUID parameter or bucket name eg. https://github.com/geonetwork/core-geonetwork/blob/develop/services/src/main/java/org/fao/geonet/api/processing/XslProcessApi.java#L120-L131 , and remove "selection" from the requestMapping ?